### PR TITLE
Add VRF support to base traceroute and ping

### PR DESCRIPTION
**Pull request by XioNoX**
Cf. https://github.com/napalm-automation/napalm-ios/pull/102 as well.

Please let me know if anything is wrong.

### CODE REVIEW

File: napalm_base/base.py [hunk 1]
```diff
@@ -1178,7 +1178,8 @@ def traceroute(self,
                    destination,
                    source=c.TRACEROUTE_SOURCE,
                    ttl=c.TRACEROUTE_TTL,
-                   timeout=c.TRACEROUTE_TIMEOUT):
+                   timeout=c.TRACEROUTE_TIMEOUT,
+                   vrf=C.TRACEROUTE_VRF):
```
Comment (mirceaulinic) [inline napalm_base/base.py]: `c` instead of `C`

Comment (XioNoX) [inline napalm_base/base.py]: Thanks, updated.

**Comment by dbarrosop** [comment]
Two things:

1. pylama isn not happy. Can you fix it?
2. We will have to make sure all drivers are updated before accepting this one or things will break.

**Comment by mirceaulinic** [comment]
JunOS: https://github.com/napalm-automation/napalm-junos/pull/116

**Comment by mirceaulinic** [comment]
IOS-XR: https://github.com/napalm-automation/napalm-iosxr/pull/97

**Comment by mirceaulinic** [comment]
eOS: https://github.com/napalm-automation/napalm-eos/pull/125

**Comment by mirceaulinic** [comment]
Just realised @XioNoX come with the PR for IOS: https://github.com/napalm-automation/napalm-ios/pull/102
So the only missing piece is for `pluribus` which doesn't know about VRF. So it will be just a slight adjustment in the header. And we're ready to get this in!

**Comment by mirceaulinic** [comment]
And Pluribus: https://github.com/napalm-automation/napalm-pluribus/pull/38

**Comment by mirceaulinic** [comment]
Actually we're waiting for few more drivers to implement this change: nxos, ros and vyos.

**Comment by dbarrosop** [comment]
I am a bit lost here. What's the current status? Should we tag all the related issues/PR with a `vrf` tag to be able to find them out easily?

**Comment by mirceaulinic** [comment]
ROS: https://github.com/napalm-automation/napalm-ros/pull/42

**Comment by mirceaulinic** [comment]
Vyos: https://github.com/napalm-automation/napalm-vyos/pull/12

**Comment by mirceaulinic** [comment]
@dbarrosop @XioNoX as this a API change, the vrf arg has to to added on all drivers implementing ping and/or traceroute:

- junos
- iosxr
- eos
- ios
- pluribus
- ros
- vyos

We're currently waiting for the maintainers of ros and vyos to confirm the changes I have proposed make sense.
Afterwards, we can merge this and release a new napalm-base version and one minor release for each driver.

**Pull request merged**